README.md
# NAME

WWW::KrispyKreme::HotLight - Fetch Krispy Kreme locations near a given geolocation!

# SYNOPSIS

    use WWW::KrispyKreme::HotLight;

    my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

    # get array ref of all the krispy kreme locations near given geo
    my $stores = $donuts->locations;

    foreach my $store (@$stores) {

        # boolean value which shows if the Hot Light is on now!
        my $is_fresh = $store->{Hotlight};

        # Does this store serve coffee?
        my $has_coffee = $store->{OffersCoffee};

        # And WiFi?
        my $has_wifi = $store->{OffersWifi};

        # store name (Burbank, Los Angeles, etc)
        my $store_name = $store->{Name};

        # geolocation of store
        my $lat = $store->{Latitude};
        my $lon = $store->{Longitude};

        # Direct link to store page
        my $url = $store->{DetailUrl};
    }

# DESCRIPTION

WWW::KrispyKreme::HotLight is a Perl wrapper for Krispy Kreme's location search
page.  This module will return an array ref of hash refs which contain info for
all the Krispy Kreme stores near the given geolocation.

# METHODS

## new

Creates a new WWW::KrispyKreme::HotLight object.  Currently the only REQUIRED
option is 'where' and only supports geo

    my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

## locations

Returns an array ref of hash refs.  Each hash ref represents a store near the
given geolocation.  The structure of one location entry should look like this:

    {
        "Location" => {
            "Id"              => 204,
            "LocationNumber"  => 1020,
            "Name"            => "Mountain View",
            "Slug"            => "mountain-view",
            "DetailUrl"       => "http://krispykreme.com/location/mountain-view",
            "LocationType"    => "Franchise",
            "Address1"        => "2146 Leghorn Street",
            "Address2"        => "",
            "City"            => "Mountain View",
            "Province"        => "CA",
            "PostalCode"      => "94043",
            "Country"         => "US",
            "PhoneNumber"     => "(650) 254-1231",
            "Latitude"        => 37.419002,
            "Longitude"       => -122.094216,
            "FundraisingType" => "Online",
            "Hotlight"        => 0,
            "OffersCoffee"    => 1,
            "OffersWifi"      => 0,
            "ExtendedDetails" => {
                "Description" => "",
                "Message" =>
                  "Hi. Thanks for stopping by your local Krispy Kreme store. Virtually, of course. Now you can see where we are and what we have going on. Oh, and be sure to LIKE us on Facebook and FOLLOW us on Twitter while you're here. It'd really mean a lot."
            },
            "Attributes" => {},
            "LocationHours" => {
                "Store Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "22:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 96,
                        "DaysOfWeekAlias" => "Fri,Sat",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "23:00:00"}]
                    }
                ],
                "Drive-Thru Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "23:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 32,
                        "DaysOfWeekAlias" => "Fri",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "00:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 64,
                        "DaysOfWeekAlias" => "Sat",
                        "Times" =>
                          [{"StartTime" => "00:00:00", "EndTime" => "00:00:00"}]
                    }
                ],
                "Hot Light Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times"           => [
                            {"StartTime" => "06:00:00", "EndTime" => "09:00:00"},
                            {"StartTime" => "17:00:00", "EndTime" => "20:00:00"}
                        ]
                    },
                    {
                        "DaysOfWeek"      => 96,
                        "DaysOfWeekAlias" => "Fri,Sat",
                        "Times"           => [
                            {"StartTime" => "06:00:00", "EndTime" => "10:00:00"},
                            {"StartTime" => "17:00:00", "EndTime" => "21:00:00"}
                        ]
                    }
                ]
            },
            "OpeningDate"    => undef,
            "OpeningDateTBD" => 0
        }
    }

# AUTHOR

Curtis Brandt &lt;curtis@cpan.org>

# COPYRIGHT

Copyright 2013- Curtis Brandt

# LICENSE

MIT license.  See LICENSE file for details.

# SEE ALSO


Changes
Revision history for WWW-KrispyKreme-Hotlight

{{$NEXT}}

1.1  2015-11-27 12:23:24 PST
        - Require Perl 5.10
        - Require Mojolicious 6.0

1.0  2015-11-24 22:49:02 PST
        - Support new API

0.06  2014-07-14 09:19:01 PDT
        - switch to a faster, all-Mojo class

0.05  2014-07-03 10:22:57 PDT
        - require Test::RequiresInternet on test
        - switch to MIT license

0.04  2013-06-15 10:11:39 PDT
        - add test: donuts.t

0.03  2013-06-11 10:41:02 PDT
        - replace mojo prereq versions as '0' to avoid cpanm version errors

0.02  2013-05-28 08:54:05 PDT
        - Rename as WWW::Krispy::Kreme::HotLight
        - Replace Moose with Mojo::Base

0.01  2013-05-17 13:23:59 PDT
        - Initial release


lib/WWW/KrispyKreme/Mojo.pm
package WWW::KrispyKreme::Mojo;

use Mojo::Base::XS -base;
use Mojo::UserAgent;

our $VERSION = '0.05';

has 'where';

has locations => \&_build_locations;

sub _build_locations {
    my ($self) = @_;
    my $geo = $self->where or return [];

    my $ua           = Mojo::UserAgent->new;
    my $base_url     = 'http://locations.krispykreme.com/Store-Locator/';
    my $hotlight_url = 'http://locations.krispykreme.com/Hotlight/HotLightStatus.ashx';
    my $header       = {'X-Requested-With' => 'XMLHttpRequest'};
    my $form         = {lat => $geo->[0], lng => $geo->[1]};

    my $locations = join ',',
      $ua->get($base_url => $header => form => $form)
      ->res->dom->find('.content-results')->attr('id')->each;

    my $json = $locations && $ua->post(
        $hotlight_url => {%$header, Referer => $base_url}
          => form => {locations => $locations}
    )->res->json;

    return $json && $json->{status} eq 'success'
      ? [@{$json->{data}{locations}}]
      : [];
}

1;
__END__

=encoding utf-8

=head1 NAME

WWW::KrispyKreme::HotLight - Fetch Krispy Kreme locations near a given geolocation!

=head1 SYNOPSIS

  use WWW::KrispyKreme::HotLight;

  my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

  # get array ref of all the krispy kreme locations near given geo
  my $stores = $donuts->locations;

  foreach my $store (@$stores) {

      # boolean value which shows if the Hot Light is on now!
      my $is_fresh = $store->{hotLightOn};

      # store name (Burbank, Los Angeles, etc)
      my $store_name = $store->{title};

      # geolocation of store
      my $geo = $store->{geoLocation};

      # shortened krispy kreme link to the location page!
      # i.e. http://kkre.me/okjGVT
      my $url = $store->{url};
  }

=head1 DESCRIPTION

WWW::KrispyKreme::HotLight is a Perl wrapper for Krispy Kreme's location search
page.  This module will return an array ref of hash refs which contain info for
all the Krispy Kreme stores near the given geolocation.

=head1 METHODS

=head2 new

Creates a new WWW::KrispyKreme::HotLight object.  Currently the only REQUIRED
option is 'where' and only supports geo

    my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

=head2 locations

Returns an array ref of hash refs.  Each hash ref represents a store near the
given geolocation.  A structure will look like this:

    [   {   'storeHours2' => '',
            'locationId'  => '993',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '818-955-9015',
            'zipcode'     => '91504',
            'state'       => 'CA',
            'city'        => 'Burbank',
            'geoLocation' => '34.190000,-118.330000',
            'url'         => 'http://kkre.me/oXMuQQ',
            'title'       => 'Burbank',
            'address'     => '1521 North Victory Place'
        },
        {   'storeHours2' => '',
            'locationId'  => '985',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '323-291-4133',
            'zipcode'     => '90008',
            'state'       => 'CA',
            'city'        => 'Los Angeles',
            'geoLocation' => '34.010000,-118.340000',
            'url'         => 'http://kkre.me/pF3cuI',
            'title'       => 'Los Angeles',
            'address'     => '4034 Crenshaw Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '4502',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-393-8319',
            'zipcode'     => '90403',
            'state'       => 'CA',
            'city'        => 'Santa Monica',
            'geoLocation' => '34.030000,-118.490000',
            'url'         => 'http://kkre.me/okjGVT',
            'title'       => 'Santa Monica',
            'address'     => '1231 Wilshire Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '984',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-532-5281',
            'zipcode'     => '90248',
            'state'       => 'CA',
            'city'        => 'Gardena',
            'geoLocation' => '33.870000,-118.290000',
            'url'         => 'http://kkre.me/r55gEc',
            'title'       => 'Gardena',
            'address'     => '1199 W. Artesia Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '989',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '626-964-5044',
            'zipcode'     => '91748',
            'state'       => 'CA',
            'city'        => 'City of Industry',
            'geoLocation' => '34.000000,-117.930000',
            'url'         => 'http://kkre.me/pnyzpu',
            'title'       => 'City Of Industry',
            'address'     => '1548 Azusa Avenue'
        }
    ]

=cut

=head1 AUTHOR

Curtis Brandt E<lt>curtis@cpan.orgE<gt>

=head1 COPYRIGHT

Copyright 2013- Curtis Brandt

=head1 LICENSE

MIT license.  See LICENSE file for details.

=head1 SEE ALSO

=cut


lib/WWW/KrispyKreme/Tiny.pm
package WWW::KrispyKreme::Tiny;

use 5.008_005;
use strict;
use warnings;

use URI;
use HTTP::Tiny;
use HTML::TreeBuilder::XPath;
use JSON::XS ();

our $VERSION = '0.05';

sub new {
    my ($class, %attr) = @_;
    bless(\%attr, $class);
}

sub locations {
    my ($self) = @_;
    my $geo = $self->{where} or return [];

    my $ua = HTTP::Tiny->new(default_headers => {'X-Requested-With' => 'XMLHttpRequest'});
    my $uri = URI->new('http://locations.krispykreme.com/Store-Locator/');
    $uri->query_form(
        {   lat => $geo->[0],
            lng => $geo->[1],
        }
    );
    #$mech->add_header('X-Requested-With' => 'XMLHttpRequest');

    my $c = $ua->get($uri->as_string);
    my $tree = HTML::TreeBuilder::XPath->new_from_content($c->{content});

    my $locations = join ',', $tree->findvalues(    #
        './/div[@class="content-locations"]//div[@class="content-results"]/@id'
    ) or return [];

    #$mech->add_header(Referer => $uri->host);
    my $status_uri = URI->new('http://locations.krispykreme.com/Hotlight/HotLightStatus.ashx');
    $c = $ua->post_form($status_uri->as_string, {locations => $locations});
    return [] unless $c->{success};

    my $json = JSON::XS::decode_json($c->{content});

    return $json && $json->{status} eq 'success'
      ? [@{$json->{data}{locations}}]
      : [];
}

1;
__END__

=encoding utf-8

=head1 NAME

WWW::KrispyKreme::HotLight - Fetch Krispy Kreme locations near a given geolocation!

=head1 SYNOPSIS

  use WWW::KrispyKreme::HotLight;

  my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

  # get array ref of all the krispy kreme locations near given geo
  my $stores = $donuts->locations;

  foreach my $store (@$stores) {

      # boolean value which shows if the Hot Light is on now!
      my $is_fresh = $store->{hotLightOn};

      # store name (Burbank, Los Angeles, etc)
      my $store_name = $store->{title};

      # geolocation of store
      my $geo = $store->{geoLocation};

      # shortened krispy kreme link to the location page!
      # i.e. http://kkre.me/okjGVT
      my $url = $store->{url};
  }

=head1 DESCRIPTION

WWW::KrispyKreme::HotLight is a Perl wrapper for Krispy Kreme's location search
page.  This module will return an array ref of hash refs which contain info for
all the Krispy Kreme stores near the given geolocation.

=head1 METHODS

=head2 new

Creates a new WWW::KrispyKreme::HotLight object.  Currently the only REQUIRED
option is 'where' and only supports geo

    my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

=head2 locations

Returns an array ref of hash refs.  Each hash ref represents a store near the
given geolocation.  A structure will look like this:

    [   {   'storeHours2' => '',
            'locationId'  => '993',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '818-955-9015',
            'zipcode'     => '91504',
            'state'       => 'CA',
            'city'        => 'Burbank',
            'geoLocation' => '34.190000,-118.330000',
            'url'         => 'http://kkre.me/oXMuQQ',
            'title'       => 'Burbank',
            'address'     => '1521 North Victory Place'
        },
        {   'storeHours2' => '',
            'locationId'  => '985',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '323-291-4133',
            'zipcode'     => '90008',
            'state'       => 'CA',
            'city'        => 'Los Angeles',
            'geoLocation' => '34.010000,-118.340000',
            'url'         => 'http://kkre.me/pF3cuI',
            'title'       => 'Los Angeles',
            'address'     => '4034 Crenshaw Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '4502',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-393-8319',
            'zipcode'     => '90403',
            'state'       => 'CA',
            'city'        => 'Santa Monica',
            'geoLocation' => '34.030000,-118.490000',
            'url'         => 'http://kkre.me/okjGVT',
            'title'       => 'Santa Monica',
            'address'     => '1231 Wilshire Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '984',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-532-5281',
            'zipcode'     => '90248',
            'state'       => 'CA',
            'city'        => 'Gardena',
            'geoLocation' => '33.870000,-118.290000',
            'url'         => 'http://kkre.me/r55gEc',
            'title'       => 'Gardena',
            'address'     => '1199 W. Artesia Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '989',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '626-964-5044',
            'zipcode'     => '91748',
            'state'       => 'CA',
            'city'        => 'City of Industry',
            'geoLocation' => '34.000000,-117.930000',
            'url'         => 'http://kkre.me/pnyzpu',
            'title'       => 'City Of Industry',
            'address'     => '1548 Azusa Avenue'
        }
    ]

=cut

=head1 AUTHOR

Curtis Brandt E<lt>curtis@cpan.orgE<gt>

=head1 COPYRIGHT

Copyright 2013- Curtis Brandt

=head1 LICENSE

MIT license.  See LICENSE file for details.

=head1 SEE ALSO

=cut


lib/WWW/KrispyKreme/Hotlight.pm
package WWW::KrispyKreme::Hotlight;

use 5.008_005;
use Mojo::Base -base;

use URI;
use WWW::Mechanize;
use HTML::TreeBuilder::XPath;
use Mojo::JSON;

our $VERSION = '0.01';

has 'where';

has locations => \&_build_locations;

sub _build_locations {
    my ($self) = @_;
    my $geo = $self->where or return [];

    my $mech = WWW::Mechanize->new;
    my $uri = URI->new('http://locations.krispykreme.com/Store-Locator/');
    $uri->query_form(
        {   lat => $geo->[0],
            lng => $geo->[1],
        }
    );
    $mech->add_header('X-Requested-With' => 'XMLHttpRequest');

    my $c = $mech->get($uri->as_string);
    my $tree = HTML::TreeBuilder::XPath->new_from_content($c->decoded_content);

    my $locations = join ',', $tree->findvalues(    #
        './/div[@class="content-locations"]//div[@class="content-results"]/@id'
    ) or return [];

    $mech->add_header(Referer => $uri->host);
    my $status_uri = URI->new('http://locations.krispykreme.com/Hotlight/HotLightStatus.ashx');
    $c = $mech->post($status_uri->as_string, Content => {locations => $locations});
    return [] unless $mech->success;

    my $json = Mojo::JSON->new->decode($c->decoded_content);

    return $json && $json->{status} eq 'success'
      ? [@{$json->{data}{locations}}]
      : [];
}

1;
__END__

=encoding utf-8

=head1 NAME

WWW::KrispyKreme::Hotlight - Fetch Krispy Kreme locations near a given geolocation!

=head1 SYNOPSIS

  use WWW::KrispyKreme::Hotlight;

  my $donuts = WWW::KrispyKreme::Hotlight->new(where => [34.170833,-118.25]);

  # get array ref of all the krispy kreme locations near given geo
  my $stores = $donuts->locations;

  foreach my $store (@$stores) {

      # boolean value which shows if the Hot Light is on now!
      my $is_fresh = $store->{hotLightOn};

      # store name (Burbank, Los Angeles, etc)
      my $store_name = $store->{title};

      # geolocation of store
      my $geo = $store->{geoLocation};

      # shortened krispy kreme link to the location page!
      # i.e. http://kkre.me/okjGVT
      my $url = $store->{url};
  }

=head1 DESCRIPTION

WWW::KrispyKreme::Hotlight is a Perl wrapper for Krispy Kreme's location search
page.  This module will return an array ref of hash refs which contain info for
all the Krispy Kreme stores near the given geolocation.

=head1 METHODS

=head2 new

Creates a new WWW::KrispyKreme::Hotlight object.  Currently the only REQUIRED
option is 'where' and only supports geo

    my $donuts = WWW::KrispyKreme::Hotlight->new(where => [34.170833,-118.25]);

=head2 locations

Returns an array ref of hash refs.  Each hash ref represents a store near the
given geolocation.  A structure will look like this:

    [   {   'storeHours2' => '',
            'locationId'  => '993',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '818-955-9015',
            'zipcode'     => '91504',
            'state'       => 'CA',
            'city'        => 'Burbank',
            'geoLocation' => '34.190000,-118.330000',
            'url'         => 'http://kkre.me/oXMuQQ',
            'title'       => 'Burbank',
            'address'     => '1521 North Victory Place'
        },
        {   'storeHours2' => '',
            'locationId'  => '985',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '323-291-4133',
            'zipcode'     => '90008',
            'state'       => 'CA',
            'city'        => 'Los Angeles',
            'geoLocation' => '34.010000,-118.340000',
            'url'         => 'http://kkre.me/pF3cuI',
            'title'       => 'Los Angeles',
            'address'     => '4034 Crenshaw Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '4502',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-393-8319',
            'zipcode'     => '90403',
            'state'       => 'CA',
            'city'        => 'Santa Monica',
            'geoLocation' => '34.030000,-118.490000',
            'url'         => 'http://kkre.me/okjGVT',
            'title'       => 'Santa Monica',
            'address'     => '1231 Wilshire Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '984',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '310-532-5281',
            'zipcode'     => '90248',
            'state'       => 'CA',
            'city'        => 'Gardena',
            'geoLocation' => '33.870000,-118.290000',
            'url'         => 'http://kkre.me/r55gEc',
            'title'       => 'Gardena',
            'address'     => '1199 W. Artesia Boulevard'
        },
        {   'storeHours2' => '',
            'locationId'  => '989',
            'hotLightOn'  => '0',
            'storeHours1' => '',
            'phone'       => '626-964-5044',
            'zipcode'     => '91748',
            'state'       => 'CA',
            'city'        => 'City of Industry',
            'geoLocation' => '34.000000,-117.930000',
            'url'         => 'http://kkre.me/pnyzpu',
            'title'       => 'City Of Industry',
            'address'     => '1548 Azusa Avenue'
        }
    ]

=cut

=head1 AUTHOR

Curtis Brandt E<lt>curtis@cpan.orgE<gt>

=head1 COPYRIGHT

Copyright 2013- Curtis Brandt

=head1 LICENSE

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 SEE ALSO

=cut


lib/WWW/KrispyKreme/HotLight.pm
package WWW::KrispyKreme::HotLight;

use Mojo::Base -base;
use Mojo::UserAgent;
use Mojo::JSON ();

our $VERSION = '1.0';

has 'where';

has locations => \&_build_locations;

sub _build_locations {
    my ($self) = @_;
    my $geo = $self->where or return [];

    my $search = {
        Where => {
            LocationTypes => ['Store', 'Commissary', 'Franchise'],
            OpeningDate => {ComparisonType => 0}
        },
        # This is important. Otherwise we receive ALL locations.
        Take => {Min => 3, DistanceRadius => 100},
        PropertyFilters =>
          {Attributes => ['FoursquareVenueId', 'OpeningType']},
    };

    my $ua           = Mojo::UserAgent->new;
    my $hotlight_url = 'http://services.krispykreme.com/api/locationsearchresult/';
    my $header       = {'Referer' => 'http://www.krispykreme.com/Locate/Location-Search'};
    my $form = {
        lat          => $geo->[0],
        lng          => $geo->[1],
        responseType => 'Full',
        search => Mojo::JSON::encode_json($search),
    };

    my $json = $ua->get(
        $hotlight_url => $header => form => $form,
    )->res->json;

    [map $_->{Location}, @$json];
}

1;
__END__

=encoding utf-8

=head1 NAME

WWW::KrispyKreme::HotLight - Fetch Krispy Kreme locations near a given geolocation!

=head1 SYNOPSIS

  use WWW::KrispyKreme::HotLight;

  my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

  # get array ref of all the krispy kreme locations near given geo
  my $stores = $donuts->locations;

  foreach my $store (@$stores) {

      # boolean value which shows if the Hot Light is on now!
      my $is_fresh = $store->{Hotlight};

      # Does this store serve coffee?
      my $has_coffee = $store->{OffersCoffee};

      # And WiFi?
      my $has_wifi = $store->{OffersWifi};

      # store name (Burbank, Los Angeles, etc)
      my $store_name = $store->{Name};

      # geolocation of store
      my $lat = $store->{Latitude};
      my $lon = $store->{Longitude};

      # Direct link to store page
      my $url = $store->{DetailUrl};
  }

=head1 DESCRIPTION

WWW::KrispyKreme::HotLight is a Perl wrapper for Krispy Kreme's location search
page.  This module will return an array ref of hash refs which contain info for
all the Krispy Kreme stores near the given geolocation.

=head1 METHODS

=head2 new

Creates a new WWW::KrispyKreme::HotLight object.  Currently the only REQUIRED
option is 'where' and only supports geo

    my $donuts = WWW::KrispyKreme::HotLight->new(where => [34.170833,-118.25]);

=head2 locations

Returns an array ref of hash refs.  Each hash ref represents a store near the
given geolocation.  The structure of one location entry should look like this:

    {
        "Location" => {
            "Id"              => 204,
            "LocationNumber"  => 1020,
            "Name"            => "Mountain View",
            "Slug"            => "mountain-view",
            "DetailUrl"       => "http://krispykreme.com/location/mountain-view",
            "LocationType"    => "Franchise",
            "Address1"        => "2146 Leghorn Street",
            "Address2"        => "",
            "City"            => "Mountain View",
            "Province"        => "CA",
            "PostalCode"      => "94043",
            "Country"         => "US",
            "PhoneNumber"     => "(650) 254-1231",
            "Latitude"        => 37.419002,
            "Longitude"       => -122.094216,
            "FundraisingType" => "Online",
            "Hotlight"        => 0,
            "OffersCoffee"    => 1,
            "OffersWifi"      => 0,
            "ExtendedDetails" => {
                "Description" => "",
                "Message" =>
                  "Hi. Thanks for stopping by your local Krispy Kreme store. Virtually, of course. Now you can see where we are and what we have going on. Oh, and be sure to LIKE us on Facebook and FOLLOW us on Twitter while you're here. It'd really mean a lot."
            },
            "Attributes" => {},
            "LocationHours" => {
                "Store Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "22:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 96,
                        "DaysOfWeekAlias" => "Fri,Sat",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "23:00:00"}]
                    }
                ],
                "Drive-Thru Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "23:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 32,
                        "DaysOfWeekAlias" => "Fri",
                        "Times" =>
                          [{"StartTime" => "06:00:00", "EndTime" => "00:00:00"}]
                    },
                    {
                        "DaysOfWeek"      => 64,
                        "DaysOfWeekAlias" => "Sat",
                        "Times" =>
                          [{"StartTime" => "00:00:00", "EndTime" => "00:00:00"}]
                    }
                ],
                "Hot Light Hours" => [
                    {
                        "DaysOfWeek"      => 31,
                        "DaysOfWeekAlias" => "Sun-Thu",
                        "Times"           => [
                            {"StartTime" => "06:00:00", "EndTime" => "09:00:00"},
                            {"StartTime" => "17:00:00", "EndTime" => "20:00:00"}
                        ]
                    },
                    {
                        "DaysOfWeek"      => 96,
                        "DaysOfWeekAlias" => "Fri,Sat",
                        "Times"           => [
                            {"StartTime" => "06:00:00", "EndTime" => "10:00:00"},
                            {"StartTime" => "17:00:00", "EndTime" => "21:00:00"}
                        ]
                    }
                ]
            },
            "OpeningDate"    => undef,
            "OpeningDateTBD" => 0
        }
    }

=cut

=head1 AUTHOR

Curtis Brandt E<lt>curtis@cpan.orgE<gt>

=head1 COPYRIGHT

Copyright 2013- Curtis Brandt

=head1 LICENSE

MIT license.  See LICENSE file for details.

=head1 SEE ALSO

=cut
